// components/popup/Dialog.js
import React from 'react'

let dialogStyles = {
    display: 'none',
    position: 'fixed',
    zIndex: '999',
    paddingTop: '100px',
    left: '50%',
    top: '50%',
    width: '100%',
    height: '100%',
    // maxWidth: '100%',
    margin: 'o auto',
    overflow: 'auto',
    position: 'fixed',
    transform: 'translate(-50%,-50%)',
    backgroundColor: 'rgb(0,0,0)',
    backgroundColor: 'rgba(0,0,0,0.4)',
    padding: '20px 12px 20px 20px',
    display: 'flex',
    //flexDirection: 'column'
}

let contentStyles = {
    backgroundColor: '#fefefe',
    margin: 'auto',
    padding: '20px',
    border: '1px solid #888',
    width: '50%'
}

let dialogCloseButtonStyles = {
    marginBottom: '15px',
    padding: '3px 8px',
    cursor: 'pointer',
    borderRadius: '50%',
    border: 'none',
    width: '30px',
    height: '30px',
    fontWeight: 'bold',
    float: 'right'
}

let textStyles = {
    display: 'flex',
    flexFlow: 'column wrap'
}
const Dialog = (props) => {

    let itemList = props.response.data.filter(({ book_id }) => book_id === props.helper)

    let dialog = (
        <div style={dialogStyles}>
            <div style={contentStyles}>
                <button
                    style={dialogCloseButtonStyles}
                    onClick={props.onClose}
                >x</button>
                <div className={textStyles}>
                    <li>Description: {itemList[0].description}</li>
                    <li>Genre: {itemList[0].genre}</li>
                    <li>Rating: {itemList[0].rating}</li>
                </div>
            </div>
        </div>
    )

    if (!props.modalShow) {
        dialog = null
    }

    return (
        <div>
            {dialog}
        </div>
    )
}

export default Dialog
